tests/models.py

```diff
@@ -1135,6 +1135,47 @@ def test_check_task_dependencies(self, trigger_rule, successes, skipped,
         self.assertEqual(completed, expect_completed)
         self.assertEqual(ti.state, expect_state)
 
+    def test_xcom_pull(self):
+        """
+        Test xcom_pull, using different filtering methods.
+        """
+        dag = models.DAG(dag_id='test_xcom', schedule_interval='@monthly')
+
+        start_date = timezone.datetime(2016, 6, 1, 0, 0, 0)
+        exec_date = timezone.utcnow()
+
+        # Push a value
+        task1 = DummyOperator(
+            task_id='test_xcom_1', dag=dag, owner='airflow',
+            start_date=start_date)
+        ti1 = TI(task=task1, execution_date=exec_date)
+        ti1.xcom_push(key='foo', value='bar')
+
+        # Push another value with the same key (but by a different task)
+        task2 = DummyOperator(
+            task_id='test_xcom_2', dag=dag, owner='airflow',
+            start_date=start_date)
+        ti2 = TI(task=task2, execution_date=exec_date)
+        ti2.xcom_push(key='foo', value='baz')
+
+        # Pull with no arguments
+        result = ti1.xcom_pull()
+        self.assertEqual(result, None)
+        # Pull the value pushed by any task. Since there are two values for
+        # "foo", it will return either (Postgres returns the last)
```

Comment:
Can we modify the code to return a consistent result across platforms?

Perhaps by adding an ORDER BY to the query?

Making it consistent, even if a bit arbitrary, seems better than having the behavior vary across platforms.

Reply:
There's already sorting happening ([here](https://github.com/bcb/incubator-airflow/blob/e7917d302b43ad37e2945fec84071bf6d40273c0/airflow/models.py#L4233)). It pulls the most recently pushed value.

In the test that works for postgres but for others it mustn't be storing the `timestamp` with enough detail to differentiate which was stored after the other. In any case, I think that's an issue outside the scope of this PR.

Reply:
@gwax this seems to be due to the use of func.now() which is database dependent.